src/go/rpk/pkg/cli/cmd/debug/bundle/bundle_linux.go

```diff
@@ -743,23 +743,144 @@ func saveDmidecode(ctx context.Context, ps *stepParams) step {
 	}
 }
 
+// fileSize is an auxiliary type that contains the path and size of a file.
+type fileSize struct {
+	path string
+	size int64
+}
+
+// walkSizeDir walks the directory tree rooted at the given path and calculates
+// the size of each file in bytes. It also excludes files whose names match
+// the given regular expression 'exclude'.
+//
+// The function returns a slice of fileSize struct, each of which contains the
+// path and size of a file, as well as the total size of the directory in bytes.
+func walkSizeDir(dir string, exclude *regexp.Regexp) (files []fileSize, size int64, err error) {
+	err = filepath.Walk(dir, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		if !info.IsDir() {
+			if !exclude.MatchString(info.Name()) {
+				size += info.Size()
+				files = append(files, fileSize{
+					path: path,
+					size: info.Size(),
+				})
+			}
+		}
+		return nil
+	})
+	return files, size, err
+}
+
+// sortControllerLogDir takes a slice of fileSize structs that represents the
+// controller logs directory and sorts it by the base_offset and term integers
+// in the filenames. Filenames should follow this format:
+// {base_offset}-{term}-{version}. If multiple files have the same base_offset,
+// then the function will sort them based on their term.
+func sortControllerLogDir(dir *[]fileSize) {
+	sort.Slice(*dir, func(i, j int) bool {
+		filename1 := filepath.Base((*dir)[i].path)
+		filename2 := filepath.Base((*dir)[j].path)
+
+		f1 := strings.Split(filename1, "-")
+		f2 := strings.Split(filename2, "-")
+
+		// One of the filenames don't follow {base_offset}-{term}-{version}.
```

Comment:
I liked the original regex version, actually -- and it allows us to define total order more easily better when a filename doesn't match what we expect.

wdyt about
* convert to regexp
* compile the regexp in a lazyInit function
* sort filenames that don't match the regex first, and if neither match the regex, sort by filename

Reply:
In which cases would the controller logs names will not match `{base_offset}-{term}-{version}`? (Other than a corrupted directory and someone manually added a file there)

I added a test case out of paranoia, but not sure if converting all to Regexp, compiling the regexp in a lazyInit is a better approach/more-readable/maintainable here.

This is what I'm imagining the code will look like:
```
func lazyRE() *regexp.Regexp {
	// Will match the controller log filename with the form
	// {base_offset}-{term}-{version}
	//   - index 0: the full match.
	//   - index 1: base_offset.
	//   - index 2: term.
	return regexp.MustCompile(`^([0-9]{1,16})-([0-9]{1,16})-v[0-9].log$`)
}

func sortControllerLogDir(dir []fileSize) {
	offsetRE := lazyRE()
	sort.Slice(dir, func(i, j int) bool {
		filename1 := filepath.Base(dir[i].path)
		filename2 := filepath.Base(dir[j].path)

		f1 := offsetRE.FindStringSubmatch(filename1)
		f2 := offsetRE.FindStringSubmatch(filename2)

		// One of the filenames don't follow {base_offset}-{term}-{version}.
		if len(f1) == 0 || len(f2) == 0 {
			return false
		}
``` 
But I don't know if I'm following what you suggested.

Reply:
I mean something like

```go
var (
    offsetRe     *regexp.Regexp
    offsetReOnce sync.Once
)

func lazyInit() {
    offsetReOnce.Do(func() {
        offsetRe = regexp.MustCompile(...)
    }
}
```

which was done frequently in Go before https://github.com/golang/go/blob/c17f8057b09b26b32f0a2aaa1efc9dd89921d431/src/internal/lazyregexp/lazyre.go was introduced

Reply:
And for the conditional:
```go
if len(f1) == 0 {
    if len(f2) == 0 {
        return f1 < f2
    }
    return true
}
if len(f2) == 0 {
    return false
}
```
which ensures any corrupted file comes first

Reply:
Got it, changed to the regexp approach but not used the lazy init since this is just called once. Thanks!